dev/com.ibm.ws.install/src/com/ibm/ws/install/internal/InstallKernelMap.java

```diff
@@ -708,4 +774,70 @@ private Integer uninstall() {
         }
         return OK;
     }
-}
+
+    private static void getShortNameFromManifest(File esa, Map<String, String> shortNameMap) throws IOException {
+        String esaLocation = esa.getAbsolutePath();
+        ZipFile zip = null;
+        try {
+            zip = new ZipFile(esaLocation);
+            Enumeration<? extends ZipEntry> zipEntries = zip.entries();
+            ZipEntry subsystemEntry = null;
+            while (zipEntries.hasMoreElements()) {
+                ZipEntry nextEntry = zipEntries.nextElement();
+                if ("OSGI-INF/SUBSYSTEM.MF".equalsIgnoreCase(nextEntry.getName())) {
+                    subsystemEntry = nextEntry;
+                    break;
+                }
+            }
+            if (subsystemEntry == null) {
+                ;
+            } else {
+                Manifest m = ManifestProcessor.parseManifest(zip.getInputStream(subsystemEntry));
+                Attributes manifestAttrs = m.getMainAttributes();
+                String shortNameAttr = manifestAttrs.getValue(SHORTNAME_HEADER_NAME);
+                shortNameMap.put(esa.getAbsolutePath(), shortNameAttr);
+            }
+        } finally {
+            if (zip != null) {
+                zip.close();
+            }
+        }
+    }
+
+    //accept List of Files
+    private File individualESAInstall(Path generatedJson, Map<String, String> shortNameMap) throws IOException, BuildException, RepositoryException {
```

Comment:
Change method name to be more descriptive, since the current name doesn't tell you what it does.  It doesn't install anything at this point.